pandas/core/reshape/concat.py

```diff
@@ -320,7 +320,8 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
 
         # Standardize axis parameter to int
         if isinstance(sample, Series):
-            axis = DataFrame()._get_axis_number(axis)
+            if not isinstance(axis, int):
+                axis = DataFrame()._get_axis_number(axis)
```

Comment:
This is a very heavy operation for a simple issue as getting the axis number.

You should try to get it done without instantiation a dataframe. Something like ``pd.DataFrame._get_axis_number(pd.DataFrame, axis)`` should probably work and be very fast.

Reply:
@topper-123 Thanks, was looking for something like that. Just pushed that change.

Reply:
Unfortunately, seems this breaks on 2.7 - I'm going to leave as is for 2.7 and only make the improvement for 3